README.md

```diff
@@ -188,6 +188,148 @@ For the deidentification of the EHR extracts, we rely on an instance running the
+- `EXPORT_AZ_CLIENT_PASSWORD`: the password, mapped to `AZURE_CLIENT_SECRET` in `docker-compose`
+- `EXPORT_AZ_TENANT_ID`: ID of the service principal's tenant. Also called its 'directory' ID. Mapped to `AZURE_TENANT_ID` in `docker-compose`
+- `EXPORT_AZ_KEY_VAULT_NAME` the name of the key vault, used to connect to the correct key vault
+
+Create the `.secrets.env` file in the _PIXL_ directory by copying the sample:
+
+```bash
+cp .secrets.env.sample .secrets.env
+```
+
+and fill in the missing values.
+
+<details><summary>Azure Keyvault setup</summary>
+
+## Azure Keyvault setup
+
+_This is done for the \_UCLH_DIF_ `dev` tenancy, will need to be done once in the _UCLHAZ_ `prod`
+tenancy when ready to deploy to production.\_
+
+This Key Vault and secret must persist any infrastructure changes so should be separate from disposable
+infrastructure services. ServicePrincipal is required to connect to the Key Vault.
+
+The application uses the ServicePrincipal and password to authenticate with Azure via environment
+variables. See [here](https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.environmentcredential?view=azure-python)
+for more info.
+
+The Key Vault and ServicePrincipal have already been created for the `dev` environment and details
+are stored in the `pixl-secrets` note in the shared PIXL folder on _LastPass_.
+
+The process for doing so using the `az` CLI tool is described below.
+This process must be repeated for `staging` & `prod` environments.
+
+### Step 1
+
+Create the Azure Key Vault in an appropriate resource group:
+
+```bash
+az keyvault create --resource-group <resource-group-name> --name <key-vault-name> --location "UKSouth"
+```
+
+### Step 2
+
+Create Service Principal & grant access as per
+
+```bash
+az ad sp create-for-rbac -n pixl-secrets --skip-assignment
+```
+
+This will produce the following output
+
+```json
+{
+    "appId": "<generated-app-ID>",
+    "displayName": "<app-name>",
+    "name": "http://<app-name>",
+    "password": "<generated-password>",
+    "tenant": "<tenant-ID>"
+}
+```
+
+### Step 3
+
+Assign correct permissions to the newly created ServicePrincipal
+
+```bash
+az keyvault set-policy --name <key-vault-name> --spn <generated-app-ID> --secret-permissions backup delete get list set
+```
+
+### Step 4
+
+Create a secret and store in the Key Vault
+
+Use Python to create a secret:
+
+```python
+import secrets
+secrets.token_urlsafe(32)
+```
+
+copy the secret and paste as <secret-value> below
+
+```bash
+az keyvault secret set --vault-name "<key-vault-name>" --name "<secret-name>" --value "<secret-value>"
+```
+
+### Step 5
+
+Save credentials in `.secrets.env` and a LastPass `PIXL Keyvault <project-slug> secrets` note.
+
+```
+EXPORT_AZ_CLIENT_ID=<generated-app-ID>
+EXPORT_AZ_CLIENT_PASSWORD=<generated-password>
+EXPORT_AZ_TENANT_ID=<tenant-ID>
+EXPORT_AZ_KEY_VAULT_NAME=<key-vault-name>
+```
+</details>
```

Comment:
At a quick search I don't think we've done this step, can you add it to lastpass?

Reply:
Yes we have, it's called pixl_screts.env

Reply:
okay, can we change the documentation do reflect this, because its not to do with the <project-slug>. Maybe call it `pixl_dev_secrets.env`?

Reply:
Ah yes I updated the original `pixl-secrets` note with this, as it was causing some confusion.
But I can create a proper project note as well.